[2023-07-11T10:45:46.927+0000] {taskinstance.py:1103} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: thyroid_preprocessing_DAG.fit_ordinal_imputer manual__2023-07-11T10:43:42.703548+00:00 [queued]>
[2023-07-11T10:45:46.976+0000] {taskinstance.py:1103} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: thyroid_preprocessing_DAG.fit_ordinal_imputer manual__2023-07-11T10:43:42.703548+00:00 [queued]>
[2023-07-11T10:45:46.978+0000] {taskinstance.py:1308} INFO - Starting attempt 1 of 1
[2023-07-11T10:45:47.048+0000] {taskinstance.py:1327} INFO - Executing <Task(PythonOperator): fit_ordinal_imputer> on 2023-07-11 10:43:42.703548+00:00
[2023-07-11T10:45:47.112+0000] {standard_task_runner.py:57} INFO - Started process 963 to run task
[2023-07-11T10:45:47.126+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'thyroid_preprocessing_DAG', 'fit_ordinal_imputer', 'manual__2023-07-11T10:43:42.703548+00:00', '--job-id', '465', '--raw', '--subdir', 'DAGS_FOLDER/preprocess_dagfile.py', '--cfg-path', '/tmp/tmpm96g5937']
[2023-07-11T10:45:47.130+0000] {standard_task_runner.py:85} INFO - Job 465: Subtask fit_ordinal_imputer
[2023-07-11T10:45:47.476+0000] {task_command.py:410} INFO - Running <TaskInstance: thyroid_preprocessing_DAG.fit_ordinal_imputer manual__2023-07-11T10:43:42.703548+00:00 [running]> on host 93aa310e5d41
[2023-07-11T10:45:47.976+0000] {taskinstance.py:1547} INFO - Exporting env vars: AIRFLOW_CTX_DAG_EMAIL='damirbogdan39@gmail.com' AIRFLOW_CTX_DAG_OWNER='aiflow' AIRFLOW_CTX_DAG_ID='thyroid_preprocessing_DAG' AIRFLOW_CTX_TASK_ID='fit_ordinal_imputer' AIRFLOW_CTX_EXECUTION_DATE='2023-07-11T10:43:42.703548+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2023-07-11T10:43:42.703548+00:00'
[2023-07-11T10:45:48.189+0000] {xcom.py:640} ERROR - Object of type SimpleImputer is not JSON serializable. If you are using pickle instead of JSON for XCom, then you need to enable pickle support for XCom in your *** config or make sure to decorate your object with attr.
[2023-07-11T10:45:48.194+0000] {taskinstance.py:1824} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/utils/json.py", line 91, in default
    return serialize(o)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/serialization/serde.py", line 171, in serialize
    raise TypeError(f"cannot serialize object of type {cls}")
TypeError: cannot serialize object of type <class 'sklearn.impute._base.SimpleImputer'>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 181, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 198, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/src/preprocess/imputers.py", line 19, in fit_ordinal_imputer
    ti.xcom_push(key="ordinal_imputer", value=imputer)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/utils/session.py", line 76, in wrapper
    return func(*args, session=session, **kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/taskinstance.py", line 2361, in xcom_push
    session=session,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/utils/session.py", line 73, in wrapper
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/xcom.py", line 243, in set
    map_index=map_index,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/xcom.py", line 632, in serialize_value
    return json.dumps(value, cls=XComEncoder).encode("UTF-8")
  File "/usr/local/lib/python3.7/json/__init__.py", line 238, in dumps
    **kw).encode(obj)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/utils/json.py", line 104, in encode
    return super().encode(o)
  File "/usr/local/lib/python3.7/json/encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/usr/local/lib/python3.7/json/encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/utils/json.py", line 93, in default
    return super().default(o)
  File "/usr/local/lib/python3.7/json/encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type SimpleImputer is not JSON serializable
[2023-07-11T10:45:48.236+0000] {taskinstance.py:1350} INFO - Marking task as FAILED. dag_id=thyroid_preprocessing_DAG, task_id=fit_ordinal_imputer, execution_date=20230711T104342, start_date=20230711T104546, end_date=20230711T104548
[2023-07-11T10:45:48.279+0000] {standard_task_runner.py:109} ERROR - Failed to execute job 465 for task fit_ordinal_imputer (Object of type SimpleImputer is not JSON serializable; 963)
[2023-07-11T10:45:48.322+0000] {local_task_job_runner.py:225} INFO - Task exited with return code 1
[2023-07-11T10:45:48.467+0000] {taskinstance.py:2653} INFO - 0 downstream tasks scheduled from follow-on schedule check
